sympy/solvers/solveset.py

```diff
@@ -3445,8 +3445,11 @@ def _separate_poly_nonpoly(system, symbols):
     denominators = set()
     poly = None
     for eq in system:
-        # Store denom expression if it contains symbol
+        # Store denom expressions that contain symbols
         denominators.update(_simple_dens(eq, symbols))
+        # Convert equality to expression
+        if isinstance(eq, Equality):
+            eq = eq.rewrite(Add)
```

Comment:
I've seen this in a few places in the codebase. Why is `rewrite(Add)` preferred over `lhs - rhs`?

To me rewriting an equation as an expression is not even really a "rewrite" since usually rewrite in sympy returns a symbolic object that is equivalent to the original.

Reply:
It's just a guaranteed way to know you have done the right thing: `eq.rewrite(Add)` vs `eq.rhs - eq.rhs`.

Reply:
(Historically, the option to write `eq.as_expr()` was frowned upon so the more specific `rewrite as an Add` was added.)